lib/trento/activity_logging/parser/metadata_enricher.ex

```diff
@@ -79,7 +69,31 @@ defmodule Trento.ActivityLog.Logger.Parser.MetadataEnricher do
   defp detect_enrichment(:cluster, {_, %{cluster_id: id}}), do: {:ok, id}
   defp detect_enrichment(:database, {_, %{database_id: id}}), do: {:ok, id}
   defp detect_enrichment(:sap_system, {_, %{sap_system_id: id}}), do: {:ok, id}
+  defp detect_enrichment(:user, {_, %{user_id: id}}), do: {:ok, id}
+
+  defp detect_enrichment(
+         target_entity,
+         {activity,
+          %{
+            resource_id: resource_id,
+            resource_type: target_entity
+          }}
+       )
+       when activity in [:resource_tagging, :resource_untagging],
+       do: {:ok, resource_id}
 
   defp detect_enrichment(_target_entity, {_activity, _metadata}),
     do: {:error, :no_enrichment_needed}
+
+  defp polish_entity(%{username: username, deleted_at: deleted_at} = entity) do
```

Comment:
I thought you would add this to the `Users.by_id/1` 👀 However, as you wish.

Reply:
Well, huge no. I won't add a custom patch to a context. I'm already ashamed to add a patch at this level, but I can swallow it.

Whether the data is "dirty" or not must be decided by the context, in a dedicated discussion hopefully

Reply:
Ok, I see. Fine with this, thanks 👍